var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -1,88 +1,125 @@
+    # and to add it as a standalone target.
+    @when('@:6.0.0')
+    def patch(self):
+        makefile_path = os.path.join('src', 'Makefile')
+        with open(makefile_path, 'r') as makefile:
+            esmumps_enabled = any(re.search(r'^esmumps(\s*):(.*)$', line) for line in makefile.readlines())
 
-        makefile_inc.append('CCS       = $(CC)')
+        if not esmumps_enabled:
+            mff = FileFilter(makefile_path)
+            mff.filter(r'^.*((esmumps)|(ptesmumps)).*(install).*$', '')
 
-        if '+mpi' in self.spec:
+            makefile_esmumps_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'Makefile.esmumps')
+            with open(makefile_path, 'a') as makefile:
+                makefile.write('\ninclude %s\n' % makefile_esmumps_path)
+
+    @when('@6.0.1:')
+    def patch(self):
+        pass
+
+    # NOTE: Configuration of Scotch is achieved by writing a 'Makefile.inc' file
+    # that contains all of the configuration variables and their desired values
+    # for the installation.  This function writes this file based on the given
+    # installation variants.
+    def configure(self):
+        makefile_inc = []
+        cflags = [
+            '-O3',
+            '-DCOMMON_RANDOM_FIXED_SEED',
+            '-DSCOTCH_DETERMINISTIC',
+            '-DSCOTCH_RENAME',
+            '-DIDXSIZE64'
+            ]
+
+        ## Library Build Type ##
+
+        if '+shared' in self.spec:
             makefile_inc.extend([
-                    'CCP       = %s' % os.path.join(self.spec['mpi'].prefix.bin, 'mpicc'),
-                    'CCD       = $(CCP)'
-                    ])
+                'LIB       = .so',
+                'CLIBFLAGS = -shared -fPIC',
+                'RANLIB    = echo',
+                'AR	       = $(CC)',
+                'ARFLAGS   = -shared $(LDFLAGS) -o'
+                ])
+            cflags.append('-fPIC')
         else:
             makefile_inc.extend([
-                    'CCP       = mpicc', # It is set but not used
-                    'CCD       = $(CCS)'
-                    ])
+                'LIB       = .a',
+                'CLIBFLAGS = ',
+                'RANLIB    = ranlib',
+                'AR	       = ar',
+                'ARFLAGS   = -ruv '
+                ])
 
+        ## Compiler-Specific Options ##
 
+        if self.compiler.name == 'gcc':
+            cflags.append('-Drestrict=__restrict')
+        elif self.compiler.name == 'intel':
+            cflags.append('-restrict')
 
-    def library_build_type(self, makefile_inc, defines):
-        makefile_inc.extend([
-            'LIB       = .a',
-            'CLIBFLAGS = ',
-            'RANLIB    = ranlib',
-            'AR	       = ar',
-            'ARFLAGS   = -ruv '
-            ])
+        makefile_inc.append('CCS       = $(CC)')
+        makefile_inc.append('CCP       = %s' %
+            (os.path.join(self.spec['mpi'].prefix.bin, 'mpicc') if '+mpi' in self.spec else 'mpicc'))
```

Comment:
@davydden: Good call.  I updated this package to use the improved syntax with my most recent change.